### PR TITLE
Initialize class variables with default values in servers/

### PR DESCRIPTION
**Pull request by valdaarhun**
Part of #43636 

As of now, I have only finished working on the files in the `servers/audio/` directory.

### CODE REVIEW

**Comment by valdaarhun** [comment]
As of now, I have only finished working on the files in the `servers/audio/` directory.

**Comment by valdaarhun** [comment]
The checks seem to have failed because of certain changes in the code. I am going to start working on fixing them now.

File: servers/audio/audio_rb_resampler.h [hunk 1]
```diff
@@ -36,26 +36,26 @@
...
-	uint32_t read_buff_len;
-	uint32_t channels;
-	uint32_t src_mix_rate;
-	uint32_t target_mix_rate;
-
-	volatile int rb_read_pos;
-	volatile int rb_write_pos;
-
-	int32_t offset; //contains the fractional remainder of the resampler
+	uint32_t rb_bitsi = 0;
```
Comment (qarmin) [inline servers/audio/audio_rb_resampler.h:39]: `rb_bits` should be instead `rb_bitsi`

File: servers/rendering/rendering_server_globals.h [hunk 2]
```diff
@@ -41,13 +41,13 @@ class RendererScene;
 
 class RenderingServerGlobals {
 public:
-	static RendererStorage *storage;
-	static RendererCanvasRender *canvas_render;
-	static RendererCompositor *rasterizer;
+	static RendererStorage *storage = nullptr;
+	static RendererCanvasRender *canvas_render = nullptr;
+	static RendererCompositor *rasterizer = nullptr;
```
Comment (qarmin) [inline servers/rendering/rendering_server_globals.h:46]: Static variables shouldn't be initialized in this way.
For now it is better to just remove initialization in this lines

Comment (valdaarhun) [reply servers/rendering/rendering_server_globals.h:46]: Yeah. I learnt this right now while going through the build error logs. I am correcting them right now. Sorry.

File: servers/audio/audio_rb_resampler.h (hunk 1 above)
Comment (valdaarhun) [reply servers/audio/audio_rb_resampler.h]: I have changed it to `rb_bits` now.

File: servers/text_server.h [hunk 3]
```diff
@@ -414,8 +414,8 @@ class TextServerManager : public Object {
 		void *user_data = nullptr;
 	};
 
-	static TextServerCreate server_create_functions[MAX_SERVERS];
-	static int server_create_count;
+	static TextServerCreate server_create_functions[MAX_SERVERS] = {};
+	static int server_create_count = 0;
```
Comment (bruvzg) [inline servers/text_server.h:418]: These are static variables, this won't build. Static vars are initialized in the `.cpp`.

Comment (valdaarhun) [reply servers/text_server.h:418]: Yeah, I am working on fixing them right now. Sorry.

**Comment by valdaarhun** [comment]
I think I have finished initializing all the variables in the `servers/` directory. I have tried to fix the things that were failing the build checks. However, there are still a few errors in the build logs but I am not able to understand what's wrong and how I am supposed to fix them.
There are three errors in the `linux build logs` related to the "use of deleted functions". I haven't understood what that means. I tried googling for info but couldn't find anything that could help me over here.

**Comment by qarmin** [comment]
Looks that if we use zeromem function, then we can't initialize any value in struct/class.
This patch fixes issues for me - [patch.patch.txt](https://github.com/godotengine/godot/files/5852137/patch.patch.txt)

File: servers/rendering/renderer_rd/renderer_scene_render_rd.h [hunk 4]
```diff
@@ -205,25 +205,25 @@ class RendererSceneRenderRD : public RendererSceneRender {
 	} sky_shader;
 
 	struct SkyShaderData : public RendererStorageRD::ShaderData {
-		bool valid;
+		bool valid = false;
 		RID version;
 
-		PipelineCacheRD pipelines[SKY_VERSION_MAX];
+		PipelineCacheRD pipelines[SKY_VERSION_MAX] = {};
```
Comment (qarmin) [inline servers/rendering/renderer_rd/renderer_scene_render_rd.h:211]: This must be reverted

File: servers/rendering/renderer_rd/renderer_scene_render_rd.h [hunk 5]
```diff
@@ -387,55 +387,55 @@ class RendererSceneRenderRD : public RendererSceneRender {
...
+		uint32_t rect_pos[2] = {};
+		uint32_t rect_size[2] = {};
+		uint32_t prev_rect_ofs[2] = {};
+		uint32_t prev_rect_size[2] = {};
+		uint32_t flip_x = 0;
+		uint32_t flip_y = 0;
+		float dynamic_range = 0.0;
+		uint32_t on_mipmap = 0;
+		float propagation = 0.0;
+		float pad[3] = {};
```
Comment (qarmin) [inline servers/rendering/renderer_rd/renderer_scene_render_rd.h:438]: Also this

File: servers/rendering/renderer_rd/renderer_scene_render_rd.h [hunk 6]
```diff
@@ -1040,34 +1034,34 @@ class RendererSceneRenderRD : public RendererSceneRender {
...
+			int32_t step_size = 0;
 
-			int32_t half_size;
-			uint32_t occlusion_index;
-			int32_t cascade;
-			uint32_t pad;
+			int32_t half_size = 0;
+			uint32_t occlusion_index = 0;
+			int32_t cascade = 0;
+			uint32_t pad = 0;
```
Comment (qarmin) [inline servers/rendering/renderer_rd/renderer_scene_render_rd.h:1046]: And this

File: servers/rendering/renderer_rd/renderer_storage_rd.h [hunk 7]
```diff
@@ -1172,31 +1149,31 @@ class RendererStorageRD : public RendererStorage {
...
 
 		struct Value {
-			float x;
-			float y;
-			float z;
-			float w;
+			float x = 0.0;
+			float y = 0.0;
+			float z = 0.0;
+			float w = 0.0;
```
Comment (qarmin) [inline servers/rendering/renderer_rd/renderer_storage_rd.h:1162]: And this also

**Review by qarmin — CHANGES_REQUESTED** [review]

**Comment by valdaarhun** [comment]
I have applied the patch. Let's hope the CI goes green.

**Comment by akien-mga** [comment]
I have yet to do an in-depth review (will try to do so ASAP), but note that before this can be merged, the commits should all be squashed into one (instead of 40 currently). See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase).

**Comment by valdaarhun** [comment]
Hi. I am really sorry for the late response. I have been very busy with college work for the last one month. I'll try to complete this issue ASAP.

**Comment by valdaarhun** [comment]
I have fixed all the conflicts in this branch while there are still conflicts in PR [#46602](https://github.com/godotengine/godot/pull/46602). Should I continue working on this branch or should I wait for the conflicts to get resolved in the other branch?

**Comment by valdaarhun** [comment]
Closed this issue in accordance with PR #46602.

**Pull request closed without merging**